src/main/kotlin/no/nav/amt/deltaker/bff/innbygger/InnbyggerApi.kt

```diff
@@ -0,0 +1,39 @@
+package no.nav.amt.deltaker.bff.innbygger
+
+import io.ktor.server.application.call
+import io.ktor.server.auth.authenticate
+import io.ktor.server.response.respond
+import io.ktor.server.routing.Routing
+import io.ktor.server.routing.get
+import no.nav.amt.deltaker.bff.application.plugins.getPersonident
+import no.nav.amt.deltaker.bff.auth.TilgangskontrollService
+import no.nav.amt.deltaker.bff.deltaker.DeltakerService
+import no.nav.amt.deltaker.bff.deltaker.model.Deltaker
+import no.nav.amt.deltaker.bff.innbygger.model.InnbyggerDeltakerResponse
+import no.nav.amt.deltaker.bff.innbygger.model.toInnbyggerDeltakerResponse
+import no.nav.amt.deltaker.bff.navansatt.NavAnsattService
+import no.nav.amt.deltaker.bff.navansatt.navenhet.NavEnhetService
+import java.util.UUID
+
+fun Routing.registerInnbyggerApi(
+    deltakerService: DeltakerService,
+    tilgangskontrollService: TilgangskontrollService,
+    navAnsattService: NavAnsattService,
+    navEnhetService: NavEnhetService,
+) {
+    fun komplettInnbyggerDeltakerResponse(deltaker: Deltaker): InnbyggerDeltakerResponse {
+        val ansatte = navAnsattService.hentAnsatteForDeltaker(deltaker)
+        val enhet = deltaker.vedtaksinformasjon?.sistEndretAvEnhet?.let { navEnhetService.hentEnhet(it) }
+        return deltaker.toInnbyggerDeltakerResponse(ansatte, enhet)
+    }
+
+    authenticate("INNBYGGER") {
+        get("/innbygger/{id}") {
+            val innbygger = getPersonident()
+            val deltaker = deltakerService.get(UUID.fromString(call.parameters["id"])).getOrThrow()
```

Comment:
Egentlig synes jeg det er god praksis å gjøre kallet videre ned til databasen med både id og fnr som parametre for å være helt sikker på at man ikke kan hente data som gjelder andre enn seg selv. Vi sjekker forsåvidt tilgangen uansett, men jeg synes det gir ekstra trygghet.

Reply:
Jeg er usikker på hvilken trygghet det gir, sjekken poao-tilgang gjør er uansett om fnr == fnr, og jeg tenker det er riktigere at det er poao-tilgang (eller om vi gjør det selv) som avgjør hvem som har tilgang til brukeren (hvis vi må legge opp til at man skal kunne gi andre fullmakt til å ta avgjørelser).

Reply:
Hvis man designer databasespørringene slik at man uansett kun kan hente egne data så gir det innebygget tilgangskontroll. Da hadde vi egentlig ikke trengt kallet til poao-tilgang, og det hadde heller ikke vært krise om det ble glemt. Risikoen for at vi glemmer å legge til tilgangskontroll som sjekker at man kun får tilgang til egne data på et api er jo til stede. Derfor har det vært vanlig praksis på alle team jeg har vært på som har innbyggerrettede tjenester at man bruker fnr fra token til å gjøre spørringer i databasen med. Men jeg se poenget med å støtte fullmakt - og da funker det jo ikke å ha fnr med ned i databasespørringen.

Reply:
Ser den, det er en smart løsning!